sumatra/core.py

```diff
@@ -105,8 +125,67 @@ def register(self, component):
                     name = component.name
                 else:
                     name = component.__name__
-                self.components[base_class][name] = component
+                self._components[base_class][name] = component
                 return
         raise TypeError("%s is not a Sumatra component." % component)
 
-registry = Registry()
+
+def component_type(cls):
```

Comment:
I think the name is ok. I guess we can catch mistaken uses by checking that a `@component_type` decorated class is not a subclass of an existing component type.

In any case it is part of the public interface of Sumatra, in my opinion (assuming that it becomes possible for plugins to define new component types in future), so it should not have a leading underscore.